docs/data-sources/ddm_instances.md

```diff
@@ -0,0 +1,97 @@
+---
+subcategory: "Distributed Database Middleware (DDM)"
+---
+
+# huaweicloud_ddm_instances
+
+Use this data source to get the list of DDM instances.
+
+## Example Usage
+
+```hcl
+resource "huaweicloud_vpc" "test" {
+  name = "test_vpc"
+  cidr = "192.168.0.0/24"
+}
+
+resource "huaweicloud_vpc_subnet" "test" {
+  name       = "test_subnet"
+  cidr       = "192.168.0.0/24"
+  gateway_ip = "192.168.0.1"
+  vpc_id     = huaweicloud_vpc.test.id
+}
+
+resource "huaweicloud_networking_secgroup" "test" {
+  name = "test_secgroup"
+}
+
+data "huaweicloud_availability_zones" "test" {}
+
+data "huaweicloud_ddm_engines" test {
+  version = "3.0.8.5"
+}
+
+data "huaweicloud_ddm_flavors" test {
+  engine_id = data.huaweicloud_ddm_engines.test.engines[0].id
+  cpu_arch  = "X86"
+}
+
+resource "huaweicloud_ddm_instance" "test" {
+  name              = "ddm-test"
+  flavor_id         = data.huaweicloud_ddm_flavors.test.flavors[0].id
+  node_num          = 2
+  engine_id         = data.huaweicloud_ddm_engines.test.engines[0].id
+  vpc_id            = huaweicloud_vpc.test.id
+  subnet_id         = huaweicloud_vpc_subnet.test.id
+  security_group_id = huaweicloud_networking_secgroup.test.id
+
+  availability_zones = [
+    data.huaweicloud_availability_zones.test.names[0]
+  ]
+}
+
```

Comment:
the above codes can be deleted, you can use variables to instead them

Reply:
done